src/BuiltInTools/AspNetCoreDeltaApplier/StartupHook.cs

```diff
@@ -0,0 +1,142 @@
+using System.IO.Pipes;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Watcher.Tools;
+
+internal sealed class StartupHook
+{
+    private static readonly bool LogDeltaClientMessages = Environment.GetEnvironmentVariable("HOTRELOAD_DELTA_CLIENT_LOG_MESSAGES") == "1";
+
+    public static void Initialize()
+    {
+        Task.Run(async () =>
+        {
+            try
+            {
+                await ReceiveDeltas();
+            }
+            catch (Exception ex)
+            {
+                Log(ex.Message);
+            }
+        });
+    }
+
+    private static List<Action> GetAssembliesReceivingDeltas()
+    {
+        var receipients = new List<Action>();
+
+        foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+        {
+            var customAttributes = assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            var deltaReceiverAttributes = customAttributes.Where(a => a.Key == "ReceiveHotReloadDeltaNotification" && !string.IsNullOrEmpty(a.Value));
+            foreach (var deltaReceiverAttribute in deltaReceiverAttributes)
+            {
+                Log($"Attempting to locate receiver {deltaReceiverAttribute.Value} in assembly {assembly}");
+                var type = assembly.GetType(deltaReceiverAttribute.Value!, throwOnError: false);
+                if (type is null)
+                {
+                    Log($"Could not find delta receiver type {deltaReceiverAttribute.Value} in assembly {assembly}.");
+                    continue;
+                }
+
+                if (type.GetMethod("DeltaApplied", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static) is MethodInfo methodInfo)
+                {
+                    var action = methodInfo.CreateDelegate<Action>();
+                    Action safeAction = () =>
+                    {
+                        try 
+                        { 
+                            action(); 
+                        } 
+                        catch (Exception ex) 
+                        { 
+                            Log(ex.ToString()); 
+                        }
+                    };
+
+                    receipients.Add(safeAction);
+                }
+                else
+                {
+                    Log($"Could not find method 'DeltaApplied' on type {type}.");
+                }
+            }
+        }
+
+        return receipients;
+    }
+
+    public static async Task ReceiveDeltas()
+    {
+        Log("Attempting to receive deltas.");
+
+        using var pipeClient = new NamedPipeClientStream(".", "netcore-hot-reload", PipeDirection.InOut, PipeOptions.CurrentUserOnly | PipeOptions.Asynchronous);
+        try
+        {
+            await pipeClient.ConnectAsync(5000);
+            Log("Connected.");
+        }
+        catch (TimeoutException)
+        {
+            Log("Unable to connect to hot-reload server.");
```

Comment:
Does the `hot-reload` server log out info that would help determine why it failed to connect here?

Reply:
Running `watch --verbose` should emits a fair bit of the diagnostics (for instance if the named pipe is in use), so that should probably help piecing this together.